event-based-simulators/extras/ExportProfileData.py

```diff
@@ -0,0 +1,233 @@
+    for device in deviceManagedObject:
+        deviceCount += 1
+        logger.info(f"Found device '{device.name}', id: #{device.id}, "
+                    f"owned by {device.owner}, number of children: {len(device.child_devices)}, type: {device.type}")
+        print(f"Found device '{device.name}', id: #{device.id} ")
+        logger.info(f"List of {device.name}'s child devices: ")
+        print(f"List of {device.name}'s child devices: ")
+        for childDevice in device.child_devices:
+            ExportSpecificProfileDataWithDeviceId(c8y, DATA_TYPE, createFrom, createTo, childDevice.id)
+
+    if deviceCount == 0:
+        logger.debug(f"No device in tenant {Environment.C8Y_TENANT} found")
+        print(f"No device in tenant {Environment.C8Y_TENANT} found")
+
+
+def ExportSpecificProfileDataWithDeviceId(c8y, DATA_TYPE, createFrom, createTo, DEVICE_ID):
+    deviceName = findDeviceNameById(DEVICE_ID)
+    deviceCount = 0
+    deviceExternalId, deviceExternalIdType = checkDeviceExternalIdById(DEVICE_ID)
+    if not deviceExternalId:
+        return
+    if isExternalIdHasEventBasedSimulatorProfileType(deviceExternalIdType):
+        filePath = createFilePath(fileName=deviceExternalId)
+    else:
+        return
+    logger.info(f"Search for {DATA_TYPE} data from device {deviceName}, id #{DEVICE_ID} ")
+    print(f"Search for {DATA_TYPE} data from device {deviceName}, id #{DEVICE_ID} ")
+    for device in c8y.device_inventory.select(name=deviceName):
+        deviceCount += 1
+        logger.info(f"Child device {device.name}, id #{device.id}")
+        if DATA_TYPE == "alarms":
+            jsonAlarmsList = listAlarms(c8y, device, createFrom, createTo)
+            appendDataToJsonFile(jsonAlarmsList, filePath, DATA_TYPE)
+            appendDataToJsonFile([], filePath, 'measurements')
+            logger.info(f"{DATA_TYPE.capitalize()} data is added to data file at {filePath}")
+        elif DATA_TYPE == "measurements":
+            # listing measurements of child device
+            jsonMeasurementsList = listMeasurements(c8y, device, createFrom, createTo)
+            appendDataToJsonFile(jsonMeasurementsList, filePath, DATA_TYPE)
+            appendDataToJsonFile([], filePath, 'alarms')
+            logger.info(f"{DATA_TYPE.capitalize()} data is added to data file at {filePath}")
+        else:
+            jsonAlarmsList = listAlarms(c8y, device, createFrom, createTo)
+            appendDataToJsonFile(jsonAlarmsList, filePath, 'alarms')
+            jsonMeasurementsList = listMeasurements(c8y, device, createFrom, createTo)
+            appendDataToJsonFile(jsonMeasurementsList, filePath, 'measurements')
+            logger.info(f"Alarms and Measurements data is added to data file at {filePath}")
+
+    if deviceCount == 0:
+        logger.debug(f"No device with id {DEVICE_ID} found")
+    return
+
+
+def findDeviceNameById(DEVICE_ID):
+    C8Y_BASE = removeSlashFromBaseUrl()
```

Comment:
Please just do this once at the beginning of the script. Instead of at every method that uses it